src/frontend/src/pages/MainPage/pages/homePage/components/McpServerTab.tsx

```diff
@@ -98,9 +225,9 @@ const McpServerTab = ({ folderName }: { folderName: string }) => {
         }, 1000);
       })
       .catch((err) => console.error("Failed to copy text: ", err));
-  };
+  }, [MCP_SERVER_JSON]);
 
-  const generateApiKey = () => {
+  const generateApiKey = useCallback(() => {
```

Comment:
I don't think this one is necessary either but i'm more ok w/ this one since the folderName doesn't change on each rerender